path_oidc_test.go

```diff
@@ -838,7 +838,8 @@ func TestOIDC_Callback(t *testing.T) {
 			t.Fatal("nil response")
 		}
 
-		if !resp.IsError() || !strings.Contains(resp.Error().Error(), `error validating signature: oidc: expected audience "abc"`) {
+		expectedMsg := "invalid id_token: one audience (not_gonna_match) which is not the client_id (abc)"
+		if !resp.IsError() || !strings.Contains(resp.Error().Error(), expectedMsg) {
```

Comment:
It looks like the library returns a specific typed error for this case, `ErrInvalidAuthorizedParty`, which you can use to check against instead.

Reply:
Hmm, I gave this a try using `errors.Is(resp.Error(), oidc.ErrInvalidAuthorizedParty)`. It returned false. Looks like it's not wrapped because of how the error is included as a string to `logical.ErrorResponse()` on [path_oidc.go#L224](https://github.com/hashicorp/vault-plugin-auth-jwt/blob/cap-oidc-lib/path_oidc.go#L224).

Reply:
That's unfortunate... We could string check against `oidc.ErrInvalidAuthorizedParty.Error()` if we wanted the test to be a bit more robust, but this is fine too.

Reply:
I like that better. Changed to check as you've suggested in d42d8d8.